### PR TITLE
chore: cache secrets-store-csi-driver v1.2.2

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
update secrets-store-csi-driver to 1.2

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
chore: cache secrets-store-csi-driver v1.2.2
```

### CODE REVIEW

**Comment by AbelHu** [comment]
@zeromagic please following the requirement for the PR
uses [conventional commit messages](https://www.conventionalcommits.org/)

**Comment by AbelHu** [comment]
I think that the release note "update secrets-store-csi-driver to 1.2"  may not be accurate. This PR only caches the new CSI images, right?

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 1]
```diff
@@ -94,8 +94,8 @@ switch ($windowsSKU) {
             "mcr.microsoft.com/oss/kubernetes-csi/azuredisk-csi:v1.20.0",
             "mcr.microsoft.com/oss/kubernetes-csi/azurefile-csi:v1.19.0",
             "mcr.microsoft.com/oss/kubernetes-csi/azurefile-csi:v1.20.0",
-            "mcr.microsoft.com/oss/azure/secrets-store/provider-azure:v1.0.0",
-            "mcr.microsoft.com/oss/kubernetes-csi/secrets-store/driver:v1.0.0",
+            "mcr.microsoft.com/oss/kubernetes-csi/secrets-store/driver:v1.2.2",
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:97]: Please keep the old ones. We should always follow below steps to upgrade versions:
1. Add new images in VHD in AgentBaker
2. Upgrade the versions in AKS
3. Remove old images in VHD in AgentBaker after the new versions are available in prod

Comment (ZeroMagic) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:97]: Due to some reasons, the current cache ones are not used in prod. So it is ok to delete them now.

**Review by AbelHu — APPROVED** [review]

**Pull request merged**